server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -2914,6 +2914,8 @@ public void testShardActiveDuringInternalRecovery() throws IOException {
         shard.openEngineAndRecoverFromTranslog();
         // Shard should now be active since we did recover:
         assertTrue(shard.isActive());
+        // Recovery state should be propagated to the engine
+        assertTrue(shard.getEngine().config().isRecoveringAsPrimary());
```

Comment:
Can we also test a recovering replica returns false?

Reply:
I've adjusted the `testShardActiveDuringInternalRecovery` test to create a primary or replica =shard randomly. Then we can verify `isRecoveringAsPrimary` based on whether we started a primary or a replica.